lib/matplotlib/tests/test_colors.py

```diff
@@ -715,3 +716,34 @@ def __add__(self, other):
         else:
             assert len(recwarn) == 0
         recwarn.clear()
+
+
+def _brighten_test_helper(color, shade, expected):
+    sc = mcolors.brighten_color(color, shade)
+    assert_equal(sc, expected)
+
+
+def test_color_brightening():
```

Comment:
I didn't write the test, so this is a disinterested comment, but: I'm still not sure what the advantage of parameterization is? I don't find the above very easy to parse...

Reply:
You don't have to explicitly loop through the cases and in this case there's no needed for the secondary helper function.

Reply:
hmmm, well, there was no need for the helper fcn anyway.  And 
```
@pytest.mark.parametrize('color, shade, expected', 
                   zip(test_colors, test_brighten, known_brighten_result)) 
```
is not any shorter than writing

```python
for color, brighten, expected in zip(test_colors,
                                      test_brighten,
                                      known_brighten_result):
```

The latter has the benefit of being standard python, whereas the former requires knowing pytest decorators.  I'm not adverse to change, but only if there is a compelling reason!

Reply:
The two big wins from parameterization come from when you can stack more than 1 and get the outer-product of the input and because it lets each test fail independently.

Reply:
I always forget about that, but the test failing independenty is the real reason to use parameterization becuase that's what makes it a true unit test. Plus in pytest you can add a list of lables for each paramterized set and so you can identify what the inputs are testing for.